### DISCUSSION TITLE
PWA issue

### DISCUSSION THREAD

**Discussion by devashish2024** [community/community#36888] [Other Feature Feedback, Questions, & Ideas] [answered]
WHen we install a PWA that is in a domain such site,github.io, it is cached. Next, when we do a commit, we need to clear cache then reinstall it. any short way?

**Comment by is-it-ayush** [comment, marked as answer]
Hey! There are two possible scenario's here:

No cache scenario (No Service worker): Your PWA app will be cached only when you use Service worker. Manifest.json will help adding your web app to home screen with a icon and open without address bar. If you don't have a service worker or if your browser don't support it, web page will be loaded fresh every single time. No Cache.

Cache-On Scenario (With Service worker): Assuming you have service workers configured, service workers can cache by lazy loading or prefetching the files that are configured for caching (You can include or exclude caching anything from html, CSS, images to JSON/XML API responses).

After the initial cache, service worker will use the cache to serve your apps network request based on [cache approach](https://developers.google.com/web/ilt/pwa/caching-files-with-service-worker) you have implemented from below ones.

- cache falling back to network
- Network falling back to cache
- Cache then network

Most apps choose to [precache](https://github.com/GoogleChromeLabs/sw-precache) due to performance benefits and look for new files on loading, if any changes found will be loaded on next session or prompt user to refresh. With this solution, each file will have a long Hash string for each file cached by service worker. On load of each application, hash code from server will be fetched to match and find what file needs to be updated and the same will be updated in a separate service worker thread. You can notice this in network tab -> service worker response in chrome developer tools.

If you choose network-first approach, you can avoid showing old content on initial load, but loose significant performance benefits that comes with caching.

So, to answer your question, "If you need to clear the cache after commit to load new content".  You can go for the precache method for specific files that you need to cache (such as assets) in your PWA to solve this issue so that it'll cache them instead of the code itself. I found this answer on [here](https://stackoverflow.com/questions/49739438/when-and-how-does-a-pwa-update-itself) and added onto it. I hope it helps.

**Comment by devashish2024** [comment]
OK
